frontend/packages/dev-console/src/components/topology/TopologyApplicationResources.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import * as _ from 'lodash';
+import { Link } from 'react-router-dom';
+import { modelFor, K8sResourceKind } from '@console/internal/module/k8s';
+import { SidebarSectionHeading } from '@console/internal/components/utils';
+import { getActiveNamespace } from '@console/internal/actions/ui';
+import TopologyApplicationResourceList from './TopologyApplicationList';
+import { TopologyDataObject } from './topology-types';
+import { getResourceDeploymentObject } from './topology-utils';
+
+import './TopologyApplicationResources.scss';
+
+const MAX_RESOURCES = 5;
+
+export type ApplicationResourceProps = {
+  title: string;
+  kind: string;
+  resourcesData: K8sResourceKind[];
+  group: string;
+};
+
+const ApplicationResource: React.FC<ApplicationResourceProps> = ({
+  title,
+  kind,
+  resourcesData,
+  group,
+}) => {
+  return (
+    <React.Fragment>
+      {!_.isEmpty(resourcesData) ? (
+        <div className="overview__sidebar-pane-body">
+          <SidebarSectionHeading text={title}>
+            {_.size(resourcesData) > MAX_RESOURCES && (
+              <Link
+                className="sidebar__section-view-all"
+                to={`/search/ns/${getActiveNamespace()}?kind=${kind}&q=${encodeURIComponent(
+                  `app.kubernetes.io/part-of=${group}`,
+                )}`}
+              >
+                {`View All (${_.size(resourcesData)})`}
+              </Link>
+            )}
+          </SidebarSectionHeading>
+          <TopologyApplicationResourceList resources={_.take(resourcesData, MAX_RESOURCES)} />
+        </div>
+      ) : null}
+    </React.Fragment>
+  );
+};
+
+export type TopologyApplicationResourcesProps = {
+  resources: TopologyDataObject[];
+  group: string;
+};
+
+const TopologyApplicationResources: React.FC<TopologyApplicationResourcesProps> = ({
```

Comment:
Same comment for this file as well. Its better to split this file into two separate files since we follow one component per file. Other than that everything looks good to me.